backend/src/apiserver/server/pipeline_server.go

```diff
@@ -573,6 +573,94 @@ func NewPipelineServer(resourceManager *resource.ResourceManager, options *Pipel
 	return &PipelineServer{resourceManager: resourceManager, httpClient: http.DefaultClient, options: options}
 }
 
+// Creates a pipeline and a pipeline version in a single transaction.
+// Supports v2beta1 behavior.
+func (s *PipelineServer) CreatePipelineAndVersion(ctx context.Context, request *apiv2beta1.CreatePipelineAndVersionRequest) (*apiv2beta1.Pipeline, error) {
+	if s.options.CollectMetrics {
+		createPipelineRequests.Inc()
+		createPipelineVersionRequests.Inc()
+	}
+
+	// URL is a required input for CreatePipelineAndVersion call
+	pipelineUrlStr := request.GetPipelineVersion().GetPackageUrl().GetPipelineUrl()
+	if pipelineUrlStr == "" {
+		return nil, util.NewInvalidInputError("Failed to create a pipeline due to missing pipeline URL")
+	}
+
+	// Get pipeline description
+	description := request.GetPipeline().GetDescription()
+	if description == "" {
+		description = request.GetPipelineVersion().GetDescription()
+	}
+
+	// Convert the input request. Fail fast if either pipeline or pipeline version is corrupted.
+	pipeline, err := toModelPipeline(request.GetPipeline())
+	if err != nil {
+		return nil, util.Wrap(err, "Failed to create a pipeline due to pipeline conversion error")
+	}
+	pipeline.Description = description
+
+	// Get pipeline name
+	displayName := request.GetPipeline().GetDisplayName()
+	if displayName == "" {
```

Comment:
Similar to above, if pipeline name is empty, should we just report an error instead of using the name of pipeline version?

Reply:
I [proposed above](https://github.com/kubeflow/pipelines/pull/9416#discussion_r1196978863) to ignore `DisplayName` and `Description` fields from `PipelineVersion` and overwrite with whatever is provided in the `Pipeline` message.